content/en/blog/2023/contributing-to-otel-demo/index.md

```diff
@@ -0,0 +1,1026 @@
+
+if (currentYear * 12 + currentMonth > year * 12 + month) {
+  throw new Error(
+    `The credit card (ending ${lastFourDigits}) expired on ${month}/${year}.`,
+  );
+}
+```
+
+The first assertion will look to validate that error message. The second
+assertion will check for the response status error code.
+
+```yaml
+# [...]
+specs:
+  - name: It should call Charge method and receive a gRPC error
+    selector:
+      span[tracetest.span.type="rpc" name="grpc.oteldemo.PaymentService/Charge"
+      rpc.system="grpc" rpc.method="Charge"
+      rpc.service="oteldemo.PaymentService"]
+    assertions:
+      - attr:grpc.error_message = "The credit card (ending 0454) expired on
+        1/2021."
+  - name: It should return a return an gRPC error code to the caller
+    selector: span[tracetest.span.type="general" name="Tracetest trigger"]
+    assertions:
+      - attr:tracetest.response.status = 2
+```
+
+[View the full test YAML here.](https://github.com/kubeshop/opentelemetry-demo/blob/main/test/tracetesting/payment-service/expired-credit-card.yaml)
+
+### Test if a Credit Card is Not Allowed
+
+[The `charge.js` has a validity check to make sure only Visa and Mastercard credit cards are allowed](https://github.com/kubeshop/opentelemetry-demo/blob/e135e81702e096ae511625a2c414e29401080633/src/paymentservice/charge.js#L62-L64).
+It will throw an error the message
+`"Sorry, we cannot process ${cardType} credit cards. Only VISA or MasterCard is accepted."`.
+
+```jsx
+if (!['visa', 'mastercard'].includes(cardType)) {
+  throw new Error(
+    `Sorry, we cannot process ${cardType} credit cards. Only VISA or MasterCard is accepted.`,
+  );
+}
+```
+
+The first assertion will look to validate that error message. The second
+assertion will check for the response status error code.
+
+```yaml
+specs:
+  - name: It should call Charge method and receive a gRPC error
+    selector:
+      span[tracetest.span.type="rpc" name="grpc.oteldemo.PaymentService/Charge"
+      rpc.system="grpc" rpc.method="Charge"
+      rpc.service="oteldemo.PaymentService"]
+    assertions:
+      - attr:grpc.error_message = "Sorry, we cannot process amex credit cards.
+        Only VISA or MasterCard is accepted."
+  - name: It should return a return an gRPC error code to the caller
+    selector: span[tracetest.span.type="general" name="Tracetest trigger"]
+    assertions:
+      - attr:tracetest.response.status = 2
+```
+
+[View the full test YAML here.](https://github.com/kubeshop/opentelemetry-demo/blob/main/test/tracetesting/payment-service/amex-credit-card-not-allowed.yaml)
+
+## What you learned about contributing to OpenTelemetry
+
+Today you learned three main things.
+
+1. How to demystify OpenTelemetry by adding automatic and manual code
+   instrumentation.
+2. How to use trace-based testing in the OpenTelemetry Demo to maintain feature
+   functionality and telemetry integrity.
+3. How to contribute to the OpenTelemetry Demo safely while avoiding
+   regressions.
```

Comment:
where did I learn that? I would have expected some words around raising a PR, etc